scripts/fetchFromSheet.ts

```diff
@@ -0,0 +1,333 @@
+  console.log(
+    chalk.red('Some Combos could not be parsed. They must be corrected in the Google Sheet')
+  );
+  comboObjects
+    .map((e, i) => [e, i] as const)
+    .filter(([e]) => !e.success)
+    .map(([e, i]) => (e.success ? (e.data as never) : ([e.error, i] as const)))
+    .map(([e, i]) =>
+      console.log(chalk.red(`[combo-${combosRawObjects[i]['id']}]:`), chalk.gray(e.toString()))
+    );
+  exit(2);
+}
+
+console.log(chalk.green(`Successfully parsed ${comboObjects.length} Combos.`));
+
+// Determing which files to purge. If they have a `#keep` comment anywhere
+// in the file at the start of a line, this Combo/Trick is kept.
+
+async function determineIfFileIsKept(fullPath: string) {
+  const yamlContent = await readFile(fullPath, 'utf-8');
+  const isKept = yamlContent
+    .split('\n')
+    .map((e) => e.trim())
+    .some((e) => e.startsWith('#keep'));
+
+  if (isKept) {
+    console.log(chalk.gray(`  File ${fullPath} is`), chalk.green(`kept`));
+  }
+  return isKept;
+}
+
+console.log(
+  chalk.blue('Deleting entries from Tricks-Folder. (keeping Tricks commented with #keep)')
+);
+
+const allTrickFiles = await globby(
+  join(fileURLToPath(import.meta.url), '../../src/data/tricks/*.yaml')
+);
+
+await Promise.all(
+  (
+    await Promise.all(
+      allTrickFiles.map(async (e) => ((await determineIfFileIsKept(e)) ? undefined : e))
+    )
+  )
+    .filter(Boolean)
+    .map(async (e) => {
+      await unlink(e!);
+      console.log(chalk.gray(`  File ${e} is`), chalk.red(`deleted`));
+    })
+);
+
+console.log(
+  chalk.blue('Deleting entries from Combos-Folder. (keeping Combos commented with #keep)')
+);
+
+const allComboFiles = await globby(
+  join(fileURLToPath(import.meta.url), '../../src/data/combos/*.yaml')
+);
+
+await Promise.all(
+  (
+    await Promise.all(
+      allComboFiles.map(async (e) => ((await determineIfFileIsKept(e)) ? undefined : e))
+    )
+  )
+    .filter(Boolean)
+    .map(async (e) => {
+      await unlink(e!);
+      console.log(chalk.gray(`  File ${e} is`), chalk.red(`deleted`));
+    })
+);
+
+// Now all "automatically tracked" Tricks / Combos are out of the repo.
+
+function createContentForEntity(data: unknown) {
+  const yamlContent = stringify(data);
+  const preamble = `
+  This file has been generated automatically. The data here is backed by a Google Sheet.
+  Do not change this by hand, as any changes will be overwritten the next time the 
+  fetchFromSheet-Script is run by a scheduled action.
+  If there are new changes that need to be synced, you can trigger the refetch by running
+  npx vite-node ./scripts/fetchFromSheet.ts from the project root. 
+  `;
+
+  return (
+    preamble
+      .split('\n')
+      .map((e) => '# ' + e.trimStart())
+      .join('\n') +
+    '\n\n' +
+    yamlContent
+  );
+}
+
+console.log(chalk.blue('Writing new Trick files...'));
+await Promise.all(
+  trickObjects
+    .map((e) => (e.success === true ? e.data : (undefined as never)))
```

Comment:
There is a small bug as it's missing `.filter(e => e.success)`

The response here is a composed type of 
```ts
type X = {
  success: true,
  data: ...
} | {
  success: false,
  error: ...
}
``` 
TS only lets me access `data` if it understands that `success` is `true`.

Unfortunately, TS does not understand that a `filter( e => e.success)` results in that condition.
![image](https://github.com/bastislack/highline-freestyle/assets/27686785/058f4faf-6e92-4879-8b6b-048638bf1a1a)

And that's where the *cursed* map comes into play. Casting a field to `never` basically tells TS that someone *never* will happen.

You usually see it on functions that will loop for ever, or can only throw something.
e.g.
![image](https://github.com/bastislack/highline-freestyle/assets/27686785/06fc8acd-4694-479e-bde5-824e06e38d66)
![image](https://github.com/bastislack/highline-freestyle/assets/27686785/1259f812-6d65-40f4-a63d-9bace323860c)

so we map the data is success, or undefined (or anything, really) if its not successful, but because we filtered for successful data, that undefined never happens — but we need to tell TS that it never will with the `never` cast

Reply:
I think I mostly understand it and this seems alright though slightly cursed. Since this is really specific please add a comment for clarity (even if it is just something as simple as "This .map acts like .filter(e => e.success) but was used for compatibility with Typescript")

Reply:
done